tests/storage/test_azure_storage.py

```diff
@@ -36,16 +36,39 @@ def test_serialize_azure_storage():
     assert serialized_storage["type"] == "Azure"
 
 
-def test_blob_service_client_property(monkeypatch):
+@pytest.mark.parametrize(
+    "secret_name,secret_arg",
+    [
+        ("SECRET_NAME", "conn_string_value"),
+        ("AZURE_STORAGE_CONNECTION_STRING", "conn_string_value"),
+    ],
+)
+def test_blob_service_client_property(monkeypatch, secret_name, secret_arg):
     connection = MagicMock()
     azure = MagicMock(from_connection_string=connection)
     monkeypatch.setattr("azure.storage.blob.BlobServiceClient", azure)
 
-    storage = Azure(container="test", connection_string="conn")
+    with context(secrets={secret_name: secret_arg}):
+        storage = Azure(container="test", connection_string_secret_name=secret_name)
+        client = storage._azure_block_blob_service()
+        azure_client = storage._azure_block_blob_service
+        assert storage.connection_string == secret_arg
 
-    azure_client = storage._azure_block_blob_service
     assert azure_client
-    connection.assert_called_with(conn_str="conn")
+    connection.assert_called_with(conn_str=secret_arg)
+
+
+@pytest.mark.parametrize(
+    "secret_name,secret_arg",
```

Comment:
I believe the test is right, but having these as the same value also opens doors to assertion being evaluated as true even if the wrong variable is wrong. I think it would be better if they are different values.

Reply:
I was wrong here. After looking it again, it shouldn't have mattered. Sorry for asking for the change.